### PR TITLE
Form+ui

### PR DESCRIPTION
**Pull request by olu144**

### CODE REVIEW

**Review by dslucas — COMMENTED** [review]
It seems like this branch isn't up to date with master. Could you run:

`git checkout master`
`git pull`
`git checkout form+UI`
`git merge master` (resolve any merge conflicts, as necessary)
`git push`

I'm happy to walk through those steps with you if it's helpful. Let us know!

**Comment by olu144** [comment]
I have ran the suggested commands and the branch is now caught up to the master.

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+@WebServlet("/contact")
+public class ContactServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
+    String email = getParameter(request, "email-input", "");
+    String phone = getParameter(request, "phone-input", "");
+    long timestamp = System.currentTimeMillis();
+    if(email.strip() == "" && phone.strip()== "") {
```
Comment (dslucas) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]:
```suggestion
    if (email.strip() == "" && phone.strip() == "") {
```

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+}
```
Comment (dslucas) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]: nit: Let's add a newline at the end of this file so this red symbol goes away.

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    Entity contactEntity = new Entity("Contact");
+    contactEntity.setProperty("Name", name);
+    contactEntity.setProperty("E-Mail", email);
+    contactEntity.setProperty("Phone", phone);
+    contactEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(contactEntity);
+    boolean emailContact = Boolean.parseBoolean(getParameter(request, "email", "false"));
+    boolean phoneContact = Boolean.parseBoolean(getParameter(request, "phone", "false"));
+    String message="";
```
Comment (dslucas) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]:
```suggestion
    String message = "";
```

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    if (!emailContact && !phoneContact){
```
Comment (dslucas) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java]:
```suggestion
    if (!emailContact && !phoneContact) {
```

File: portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    Entity contactEntity = new Entity("Contact");
+    contactEntity.setProperty("Name", name);
+    contactEntity.setProperty("E-Mail", email);
+    contactEntity.setProperty("Phone", phone);
+    contactEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(contactEntity);
+    boolean emailContact = Boolean.parseBoolean(getParameter(request, "email", "false"));
+    boolean phoneContact = Boolean.parseBoolean(getParameter(request, "phone", "false"));
```
Comment (dslucas) [inline portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java:52]: I'm curious if this is working as intended? My understanding is that `parseBoolean` returns true if the string argument is not null and is equal, ignoring case, to the string "true". For example, if the email is example@gmail.com, this boolean would be be false (but it seems like we'd want it to be true). Let me know if I'm misunderstanding the code!

Comment (olu144) [reply portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java:52]: Yes, I was able to get it working as intended. My statement receives a response from a checkbox and the "getParameter" function ensures an appropriate boolean is returned.

Comment (dslucas) [reply portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java:52]: Cool, thanks for checking!

**Review by dslucas — APPROVED** [review]

**Pull request merged**